dev/io.openliberty.io.smallrye.reactive.messaging-provider4/src/io/openliberty/microprofile/reactive/messaging30/internal/OLReactiveMessaging30Extension.java

```diff
@@ -97,20 +109,55 @@ void beforeBeanDiscovery(@Observes BeforeBeanDiscovery discovery, BeanManager be
+        final ContextBeginnerEnder contextBeginnerEnder = createContextBeginnerEnderIfCheckpointEnabled();
+        final MediatorManager mediatorManager = configureMediatorManager(beanManager);
+
+        CheckpointHook checkpointHook = new CheckpointHook() {
+
+            @Override
+            public void restore() {
+                try (ContextBeginnerEnder cbe = contextBeginnerEnder.beginContext()) {
+                    startMediatorManager(mediatorManager);
+                }
+            }
+        };
+
+        // If checkpoint is enabled this if statement will register the hook we created above, so startMediatorManager() will be called after restore. Then it returns true.
+        // If checkpoint is disabled this if statement will return false, so startMediatorManager() will be called right away.
+        if (!CheckpointPhase.getPhase().addMultiThreadedHook(checkpointHook)) {
             startMediatorManager(mediatorManager);
+        }
+    }
+
+    @Override
+    protected void startMediatorManager(MediatorManager mediatorManager) {
+        // Catch "jakarta.enterprise.inject.spi.DeploymentException: Wiring error(s) detected in application"
+        // that is thrown from MediatorManager.start() there are any validation errors with the Reactive Messaging Configuration
+        try {
+            super.startMediatorManager(mediatorManager);
         } catch (DeploymentException de) {
             StringBuilder exceptionBuilder = new StringBuilder(de.getLocalizedMessage());
             for (Throwable t : de.getSuppressed()) {
                 exceptionBuilder.append("\n" + t.getLocalizedMessage());
             }
             String appName = ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData().getJ2EEName().getApplication();
-            Tr.error(tc,  Tr.formatMessage(tc, "reactive.messaging.validation.error.CWMRX1100E", appName, exceptionBuilder.toString()));
-            // rethrow DeploymentException to make sure we stop the application correctly.
-            throw de;
+            Tr.error(tc, Tr.formatMessage(tc, "reactive.messaging.validation.error.CWMRX1100E", appName, exceptionBuilder.toString()));
+            throw de; // rethrow DeploymentException to make sure we stop the application correctly. Do it here so the compiler doesn't ask for a return statement 
+        }
+    }
+
+    private ContextBeginnerEnder createContextBeginnerEnderIfCheckpointEnabled() {
+        if (CheckpointPhase.getPhase() != CheckpointPhase.INACTIVE) {
+            Bundle b = FrameworkUtil.getBundle(ReactiveMessagingExtension.class);
+            if (b != null) {
+                BundleContext bc = b.getBundleContext();
+                ServiceReference<CDIService> sr = bc.getServiceReference(CDIService.class);
+                CDIService service = bc.getService(sr);
+                CDIRuntime cdiRuntime = (CDIRuntime) service;
+                return cdiRuntime.cloneActiveContextBeginnerEnder();
```

Comment:
Should `bc.ungetService(sr)` after you are done using the `cdiRuntime` here.  If this is called many times there is a risk of overflowing the service use count in the framework.